agent-control/src/cli/on_host/migrate_folders.rs

```diff
@@ -0,0 +1,481 @@
+use crate::agent_control::defaults::{
+    AGENT_CONTROL_ID, FOLDER_NAME_FLEET_DATA, FOLDER_NAME_LOCAL_DATA, INSTANCE_ID_FILENAME,
+    STORE_KEY_LOCAL_DATA_CONFIG, STORE_KEY_OPAMP_DATA_CONFIG,
+};
+use crate::cli::error::CliError;
+use crate::opamp::instance_id::on_host::storer::build_config_name;
+use std::fs;
+use std::path::{Path, PathBuf};
+use tracing::{debug, error};
+
+const LOCAL_DATA_DIR: &str = "/etc/newrelic-agent-control";
+const REMOTE_DATA_DIR: &str = "/var/lib/newrelic-agent-control";
+const OLD_ENV_FILE_NAME: &str = "newrelic-agent-control.conf";
+const NEW_ENV_FILE_NAME: &str = "systemd-env.conf";
+const OTEL_AGENT_ID: &str = "nrdot";
+const INFRA_AGENT_ID: &str = "nr-infra";
+// old folder and file names
+const OLD_CONFIG_AGENT_CONTROL_FILE_NAME: &str = "config.yaml";
+const OLD_IDENTIFIERS_YAML: &str = "identifiers.yaml";
+const OLD_CONFIG_SUB_AGENT_FILE_NAME: &str = "values.yaml";
+const OLD_SUB_AGENT_DATA_DIR: &str = "fleet/agents.d";
+
+/// TODO: TEMPORAL SCRIPT TO MIGRATE PATHS AND NAMES AFTER SOME TIME THIS SHOULD BE DELETED
+pub fn migrate() -> Result<(), CliError> {
+    let local_base = PathBuf::from(LOCAL_DATA_DIR);
+    let remote_base = PathBuf::from(REMOTE_DATA_DIR);
+
+    let new_local_data_path = local_base.join(FOLDER_NAME_LOCAL_DATA);
+    let new_fleet_data_path = remote_base.join(FOLDER_NAME_FLEET_DATA);
+
+    if !check_new_folders(&new_local_data_path, &new_fleet_data_path) {
+        move_and_rename(&local_base, &remote_base)?;
+    }
+    Ok(())
+}
+// Check if the new folders already exists - local-data or fleet-data
+fn check_new_folders(local_new_path: &Path, remote_new_path: &Path) -> bool {
+    let local_exists = local_new_path.exists() && local_new_path.is_dir();
+    let fleet_exists = remote_new_path.exists() && remote_new_path.is_dir();
+
+    local_exists || fleet_exists
+}
+// Copy the old files in the new paths but leaving the old ones in place
+fn move_and_rename(local_base: &Path, remote_base: &Path) -> Result<(), CliError> {
+    debug!("Starting migration: moving files from old structure to new structure...");
+
+    let migration_pairs = get_migration_list(local_base, remote_base);
+
+    for (old_path, new_path) in &migration_pairs {
+        if let Some(parent_dir) = new_path.parent() {
```

Comment:
I would add a check here to just skip the file if it does not exist. In particular whenever fleet is not eabled I fear it could "break" since there is no file and no parent.

Reply:
exactly! we exit with:
```
error!(
                "Could not determine parent directory for new path '{}'",
                new_path.display()
            );
```

I think it is a mistake, no?